### ISSUE TITLE
Regular pipe after multiple OK pipes trigger multiple function calls

### ISSUE THREAD

**Issue by Nagasaki45** [CrowdHailer/OK#37]
When an error occures in an "OK pipe" that ends with a regular elixir pipe the errored function is called multiple times.  The expected behaviour is that each function in the pipeline will be called only once.

Here's an example of the problem:

```elixir
defmodule MyModule do

  import OK, only: ["~>>": 2]

  def demo_bug do
    {:ok, :inital_value}
    ~>> failing_function()
    ~>> pass_through()
    |> summarize()
  end

  def failing_function(_whatever) do
    IO.puts "Calling failing_function"
    {:error, "failing_function always fail"}
  end

  def pass_through(value) do
    value
  end

  def summarize({:ok, _result}), do: "All good"
  def summarize({:error, error}), do: "Failed due to #{inspect(error)}"

end
```

I want the regular pipe by the end to pattern match for both errors and success conditions. If you do not intend to support this use case just say so and ignore the rest of the ticket. Otherwise, when I run the `demo_bug` function the `failing_function` is called multiple times.

```elixir
iex> MyModule.demo_bug()
Calling failing_function
Calling failing_function
"Failed due to \"failing_function always fail\""
```

The number of times the `failing_function` is called is 2 in the power of the number of "OK piped" functions that follow. If we change the code to:

```elixir
  def demo_bug do
    {:ok, :inital_value}
    ~>> failing_function()
    ~>> pass_through()
    ~>> pass_through()  # This line is new
    ~>> pass_through()  # This line is new
    |> summarize()
  end
```

The `failing_function` will be called 2 ^ 3 = 8 times!

```elixir
iex> MyModule.demo_bug()
Calling failing_function
Calling failing_function
Calling failing_function
Calling failing_function
Calling failing_function
Calling failing_function
Calling failing_function
Calling failing_function
"Failed due to \"failing_function always fail\""
```

**Comment by CrowdHailer** [comment]
I agree this is definitely the wrong behaviour.

By convention I normally separate the error bit from the final bit in your pattern

```elixir
def process do
  {:ok, :foo}
  ~>> process()
  ~>> process_more()
end

def handle_result({:ok, value}), do: "hooray"
def handle_result({:error, reason}), do: ":-("

process
|> handle_result
```

But this should not be how the library has to be used.

**Comment by CrowdHailer** [comment]
@Nagasaki45 I'm not going to get to look at this in the short term. However if you want to try a PR the `~>>` macro is only 15 lines. https://github.com/CrowdHailer/OK/blob/master/lib/ok.ex#L124-L139

**Comment by Nagasaki45** [comment]
I don't think I will have the time to look into this either. Sorry...

**Comment by CrowdHailer** [comment]
@Nagasaki45 Fixed in 1.9.2

This was the change needed 
https://github.com/CrowdHailer/OK/commit/c80af85d0e3eeba46375675a42e3764b90f68777

**Comment by Nagasaki45** [comment]
Thanks @CrowdHailer for fixing it and for letting me know. Much appreciated!
